src/main/java/io/vertx/ext/stomp/impl/SubscriptionImpl.java

```diff
@@ -0,0 +1,153 @@
+package io.vertx.ext.stomp.impl;
+
+import io.vertx.ext.stomp.Frame;
+import io.vertx.ext.stomp.StompServerConnection;
+import io.vertx.ext.stomp.Subscription;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Implementation of {@link Subscription}.
+ */
+public class SubscriptionImpl implements Subscription {
```

Comment:
A new subscription here is created every time a client subscribes to a destination, and the subscription keeps track of unacknowledged messages.

However this seems rather pointless. The purpose of message acknowledgement in STOMP is so messages MAY be delivered to other subscribers of the destination if the subscription is closed. But this never happens in this implementation so message acknowledgement appears to do nothing here ;)

STOMP itself is silent about the semantics of destinations but In other STOMP implementation (e.g. the HornetQ one) http://docs.jboss.org/hornetq/2.2.5.Final/user-manual/en/html/interoperability.html - destinations often map to queues. There can be many subscribers to a queue but only one subscriber gets each message unless it is nacked or the subscription closes.

But destinations can also mean topics.

In the current implementation the destination is assumed to be a topic (which makes acks pointless) as you always send the message to all subscriptions for the destination.

I think what is needed here is to abstract out the destination idea and provide two implementations:

1. A simple in memory queue
2. A simple pass through pub sub (which is what you do already)
3. Bridge to Vert.x eventbus (also pubsub)